src/Raven.Server/Documents/TimeSeries/TimeSeriesStorage.cs

```diff
@@ -302,6 +302,14 @@ public string DeleteTimestampRange(DocumentsOperationContext context, DeletionRa
                         return null;
                 }
 
+                if (from == DateTime.MinValue && to == DateTime.MaxValue)
+                {
+                    table.DeleteByKey(slicer.TimeSeriesKeySlice);
+                    Stats.DeleteStats(context, collectionName, slicer.StatsKey);
+                    RemoveTimeSeriesNameFromMetadata(context, slicer.DocId, slicer.Name);
```

Comment:
this will modify the document and would cause conflict, no?

Reply:
right, done :)